include/event_dbus_monitor.hpp

```diff
@@ -0,0 +1,641 @@
+
+    boost::container::flat_map<std::string, std::variant<std::string, uint8_t>>
+        values;
+    std::string objName;
+    msg.read(objName, values);
+
+    auto find = values.find("Status");
+    if (find == values.end())
+    {
+        BMCWEB_LOG_DEBUG("Status property not found. Continuing to listen...");
+        return;
+    }
+    std::string* propValue = std::get_if<std::string>(&(find->second));
+
+    if (propValue != nullptr &&
+        *propValue ==
+            "xyz.openbmc_project.Common.Progress.OperationStatus.Completed")
+    {
+        BMCWEB_LOG_DEBUG << "Sending event\n";
+
+        std::string eventOrigin;
+        // Push an event
+        if (dumpType == "bmc")
+        {
+            eventOrigin = std::format(
+                "/redfish/v1/Managers/bmc/LogServices/Dump/Entries/{}", dumpId);
+        }
+        else if (dumpType == "system")
+        {
+            eventOrigin = std::format(
+                "/redfish/v1/Systems/system/LogServices/Dump/Entries/System_{}",
+                dumpId);
+        }
+        else if (dumpType == "resource")
+        {
+            eventOrigin = std::format(
+                "/redfish/v1/Systems/system/LogServices/Dump/Entries/Resource_{}",
+                dumpId);
+        }
+        else if (dumpType == "hostboot")
+        {
+            eventOrigin = std::format(
+                "/redfish/v1/Systems/system/LogServices/Dump/Entries/Hostboot_{}",
+                dumpId);
+        }
+        else if (dumpType == "hardware")
+        {
+            eventOrigin = std::format(
+                "/redfish/v1/Systems/system/LogServices/Dump/Entries/Hardware_{}",
+                dumpId);
+        }
+        else if (dumpType == "sbe")
+        {
+            eventOrigin = std::format(
+                "/redfish/v1/Systems/system/LogServices/Dump/Entries/SBE_{}",
+                dumpId);
+        }
+        else
+        {
+            BMCWEB_LOG_ERROR("Invalid dump type received when listening for "
+                             "dump created signal");
+            return;
+        }
+        redfish::EventServiceManager::getInstance().sendEvent(
+            redfish::messages::resourceCreated(), eventOrigin, "LogEntry");
+    }
+}
+
+inline void dumpDeletedSignal(sdbusplus::message::message& msg)
+{
+    BMCWEB_LOG_DEBUG("Dump Deleted - match fired");
+
+    if (msg.is_method_error())
+    {
+        BMCWEB_LOG_ERROR("Dump Deleted signal error");
+        return;
+    }
+
+    std::vector<std::string> interfacesList;
+    sdbusplus::message::object_path objPath;
+
+    msg.read(objPath, interfacesList);
+
+    std::string dumpType;
+    std::string dumpId;
+
+    dbus::utility::getNthStringFromPath(objPath, 3, dumpType);
+    dbus::utility::getNthStringFromPath(objPath, 5, dumpId);
+
+    std::string eventOrigin;
+
+    if (dumpType == "bmc")
+    {
+        eventOrigin = "/redfish/v1/Managers/bmc/LogServices/Dump/Entries/" +
```

Comment:
I think these are also to be done with `boost::urls::format()`

Reply:
done